fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCQueryBuilder.java

```diff
@@ -314,12 +315,75 @@ public JDBCOperator getPrefixOperator(ParameterValue queryParmValue) {
+     * @return SqlQueryData - An object representing the selector query segment for the passed search parm.
+     * @throws Exception
+     * 
+     * @implNote This is just like {@link com.ibm.fhir.persistence.util.AbstractQueryBuilder.buildQueryParm(QueryParameter, String)}
+     *           except this one takes a tableAlias
+     */
+    protected SqlQueryData buildQueryParm(Class<?> resourceType, QueryParameter queryParm, String tableAlias) throws Exception {
+        final String METHODNAME = "buildQueryParm";
+        log.entering(CLASSNAME, METHODNAME, queryParm.toString());
+        
+        SqlQueryData databaseQueryParm = null;
+        Type type;
+        
+        try {
+            if (Modifier.MISSING.equals(queryParm.getModifier())) {
+                return this.processMissingParm(resourceType, queryParm, tableAlias);
+            }
+            // NOTE: The special logic needed to process NEAR and NEAR_DISTANCE query parms for the Location resource type is
+            // found in method processLocationPosition(). This method will not handle those.
+            if (! (Location.class.equals(resourceType) && 
+                (queryParm.getCode().equals(NEAR) || queryParm.getCode().equals(NEAR_DISTANCE)))) {
+                
+                type = queryParm.getType();
+                switch(type) {
+                case STRING:    databaseQueryParm = this.processStringParm(queryParm, tableAlias);
+                        break;
+                case REFERENCE: if (queryParm.isChained()) {
+                                    databaseQueryParm = this.processChainedReferenceParm(queryParm);
+                                }
+                                else if (queryParm.isInclusionCriteria()) {
+                                    databaseQueryParm = this.processInclusionCriteria(queryParm);
+                                }
+                                else {
+                                    databaseQueryParm = this.processReferenceParm(resourceType, queryParm, tableAlias);
+                                }
+                        break;
+                case DATE:      databaseQueryParm = this.processDateParm(resourceType, queryParm, tableAlias);
+                        break;
+                case TOKEN:     databaseQueryParm = this.processTokenParm(queryParm, tableAlias);
+                        break;
+                case NUMBER:    databaseQueryParm = this.processNumberParm(resourceType, queryParm, tableAlias);
+                        break;
+                case QUANTITY:  databaseQueryParm = this.processQuantityParm(resourceType, queryParm, tableAlias);
+                        break;
+                case URI:       databaseQueryParm = this.processUriParm(queryParm, tableAlias);
+                        break;
+                case COMPOSITE: databaseQueryParm = this.processCompositeParm(resourceType, queryParm, tableAlias);
+                        break;
```

Comment:
I'm expecting after the merge, there will have to be case SPECIAL:

Reply:
yep